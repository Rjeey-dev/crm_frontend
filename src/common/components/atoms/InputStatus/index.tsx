import React from 'react';

import Icon from "common/components/atoms/Icon/index";
import Span from "common/components/atoms/Span/index";

interface IProps {
    isDefault: boolean,
    isValid: boolean,
}

function InputStatus(props: IProps) {
    let classes = '';
    let wrapperClasses= 'input-group-text';

    if (props.isDefault) {
        classes = 'mdi-check-circle-outline';
    } else {
        classes = props.isValid ? 'mdi-check-circle' : 'mdi-check-circle with-error';
    }

    if (!props.isValid && !props.isDefault) {
        wrapperClasses += ' input-group-text-danger';
    }

    return <div className="input-group-append">
            <Span classes={wrapperClasses}>
                <Icon classes={'mdi ' + classes}/>
            </Span>
    </div>
}

export default InputStatus;