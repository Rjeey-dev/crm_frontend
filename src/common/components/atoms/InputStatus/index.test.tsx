import { shallow } from 'enzyme';
import React from 'react';

import InputStatus from './index';

it("Renders <InputStatus/> default and valid", () => {
    const output = shallow(
        <InputStatus isDefault={true} isValid={true}/>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <InputStatus/> default and not valid", () => {
    const output = shallow(
        <InputStatus isDefault={true} isValid={false}/>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <InputStatus/> not default and valid", () => {
    const output = shallow(
        <InputStatus isDefault={false} isValid={true}/>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <InputStatus/> not default and not valid", () => {
    const output = shallow(
        <InputStatus isDefault={false} isValid={false}/>
    ).html();

    expect(output).toMatchSnapshot();
});