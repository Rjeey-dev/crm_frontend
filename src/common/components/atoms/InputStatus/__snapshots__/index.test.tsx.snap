// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <InputStatus/> default and not valid 1`] = `"<div class=\\"input-group-append\\"><span class=\\"input-group-text input-group-text-danger\\"><i class=\\"mdi mdi-check-circle with-error\\"></i></span></div>"`;

exports[`Renders <InputStatus/> default and valid 1`] = `"<div class=\\"input-group-append\\"><span class=\\"input-group-text\\"><i class=\\"mdi mdi-check-circle-outline\\"></i></span></div>"`;

exports[`Renders <InputStatus/> not default and not valid 1`] = `"<div class=\\"input-group-append\\"><span class=\\"input-group-text input-group-text-danger\\"><i class=\\"mdi mdi-check-circle with-error\\"></i></span></div>"`;

exports[`Renders <InputStatus/> not default and valid 1`] = `"<div class=\\"input-group-append\\"><span class=\\"input-group-text\\"><i class=\\"mdi mdi-check-circle\\"></i></span></div>"`;
